util/network-devp2p/src/discovery.rs

```diff
@@ -764,6 +777,32 @@ impl<'a> Discovery<'a> {
 		Ok(None)
 	}
 
+	fn on_enr_request(&mut self, rlp: &Rlp, node_id: &NodeId, from: &SocketAddr, request_hash: &[u8]) -> Result<Option<TableUpdates>, Error> {
```

Comment:
Why isn't `request_hash` a `&H256`? Or did I misread the spec?

Reply:
I figured that the logic here is identical to `on_ping` hence the design.

Reply:
Right, I think it is designed that way because the underlying API requires slices, such as [send_packet](https://github.com/OpenEthereum/open-ethereum/blob/master/util/network-devp2p/src/discovery.rs#L422-#L428)

However, I agree with David that we should have `H256` to get more type-safety but then for the entire `API` need to be refactored, definitely out-of-scope in this PR.